nexus/nexus-strawman.md

```diff
@@ -0,0 +1,358 @@
+            log.Fatal(err)
+        }
+        defer temporalClient.Close()
+        builder, err := temporalnexus.NewWorkerBuilder(temporalnexus.WorkerBuilderOptions{Client: temporalClient})
+        if err != nil {
+            log.Fatal(err)
+        }
+        // Add workflow as ALO to Nexus
+        err = builder.AddWorkflowALOHandler("foo", temporalnexus.WorkflowALOHandlerOptions{
+            Workflow: FooWorkflow,
+            Options:  temporalclient.StartWorkflowOptions{TaskQueue: "my-task-queue"},
+        })
+        if err != nil {
+            log.Fatal(err)
+        }
+
+        // Worker definition...
+    }
+    ```
+1. Nexus is now able to process requests to its frontend endpoint.
+
+### Consuming a Nexus Service from Temporal
+
+1. User asks the Nexus Service author for the endpoint of the Nexus Service Instance they are trying to consume. For Temporal based Nexus Service Instances, the endpoint will not be HTTP or gRPC but rather a custom URI formulation.
+2. User registers the provided Nexus URI with the Temporal Server by providing an explicit mapping between canonical name and the fully qualified URI (eg: `production: temporal:72aA64A0jhOL99B2`).
+3. User calls the external Nexus Service using a Workflow-specific Nexus API which is provided by Temporal.
+    ```go
+    // calling a Nexus from a Temporal Workflow
+    func MyWorkflow(ctx workflow.Context, name string) (string, error) {
+        // Start some ALO
+        alo, err := client.ALO(ctx, temporalnexus.StartAloOptions{
+            Nexus: "production", // maps to temporal:72aA64A0jhOL99B2 because it was registered with the server
+            Name: "foo",
+            ID:   "my-id",
+            Arg:  name,
+        })
+        if err != nil {
+            return "", err
+        }
+        // Workflow implementation...
+    }
+    ```
+
+## Concepts
+
+### Nexus Instantiation
+
+Instantiating a Nexus Service makes it possible for the Nexus Backend to receive traffic meant for that specific Instance. Nexus Instantiation does make your Nexus Service Instance endpoint(s) available for traffic, but until you register and run corresponding Nexus Workers, no forward progress will be made on incoming requests. In this way, sending requests at this stage is similar to starting Workflows without any Workflow Workers running.
+
+**Note:** Meaningfully persisting the in flight Nexus Requests is not a product level requirement. Some users may not want Nexus Request payloads to be persisted and we should leave an opening for that in the design.
+
+**Registration:**
+
+For the sake of this proposal, we will only consider the case where Nexus is registered via `nexus` CLI. In the future we may evaluate and consider alternative methods of registering a Nexus Service Instance. Registering a Nexus Service Instance would be done with the following command structure:
+
+`tctl nexus register --name my-service --description "My service" --backend nexus-backend.example.com`
```

Comment:
```suggestion
`nexus register --name my-service --description "My service" --backend nexus-backend.example.com`
```

This is not Temporal specific

Reply:
I discussed with you that for the sake of this proposal I will not have Nexus as a standalone concept.